modules/standard/Buffers.chpl

```diff
@@ -352,40 +447,82 @@ module Buffers {
     return ret;
   }
 
+  /* Append a :record:`bytes` object to a :record:`buffer`.
+     This function might store the passed bytes object by reference instead of
+     copying it. The current implementation will always do so and will always
+     increase the reference count of the bytes object.  The version of this
+     function called without the error argument will halt if an error is
+     encountered.
+
+     :arg b: the :record:`bytes` object to append
+     :arg skip_bytes: how many bytes at the front of b to skip
+     :arg len_bytes: how many bytes to append to the buffer
+     :arg error: (optional) capture an error that was encountered instead of
+           halting on error
+  */
   proc buffer.append(b:bytes, skip_bytes:int(64) = 0, len_bytes:int(64) = b.len, out error:syserr) {
```

Comment:
The 'bytes' object, which typically would contain data already, is added to the buffer.
If I had a method called extend, it would add space but you wouldn't provide the pointer
to the new space to add...